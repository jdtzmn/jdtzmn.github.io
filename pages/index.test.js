import { shallowMount } from '@vue/test-utils'
import Index from '../pages/index'

describe('Index', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallowMount(Index)
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should be a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('should render a row div', () => {
    expect(wrapper.contains('div.row')).toBeTruthy()
  })

  it('should render h1 element', () => {
    expect(wrapper.contains('h1.text-heading.text-center')).toBeTruthy()
  })

  it('should render "Header" inside h1 element', () => {
    expect(wrapper.find('h1').text()).toEqual('Header')
  })

  it('should render p element', () => {
    expect(wrapper.contains('p.text-body')).toBeTruthy()
  })

  it('should render "Body" inside p element', () => {
    expect(wrapper.find('p').text()).toEqual('Body')
  })
})
