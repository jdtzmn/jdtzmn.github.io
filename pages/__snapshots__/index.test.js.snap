// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index should match snapshot 1`] = `
<div class="row">
  <h1 class="text-heading text-center">Header</h1>
  <p class="text-body">Body</p>
</div>
`;
